tests/Endpoint_selenium_unittest.py

```diff
@@ -0,0 +1,155 @@
+        # Be sure to change them before submitting a PR
+        driver.find_element_by_id("id_username").send_keys(os.environ['DD_ADMIN_USER'])
+        driver.find_element_by_id("id_password").clear()
+        driver.find_element_by_id("id_password").send_keys(os.environ['DD_ADMIN_PASSWORD'])
+        # "Click" the but the login button
+        driver.find_element_by_css_selector("button.btn.btn-success").click()
+        return driver
+
+    # first thing first. We create product, just to make sure there is atleast 1 product available
+    # to assign endpoints to when creating or editing any.
+    def test_create_product(self):
+        # Login to the site. Password will have to be modified
+        # to match an admin password in your own container
+        driver = self.login_page()
+        # Navigate to the product page
+        driver.get(self.base_url + "product")
+        # "Click" the dropdown button to see options
+        driver.find_element_by_id("dropdownMenu1").click()
+        # "Click" the add prodcut button
+        driver.find_element_by_link_text("Add Product").click()
+        # Fill in th product name
+        driver.find_element_by_id("id_name").clear()
+        driver.find_element_by_id("id_name").send_keys("QA Test")
+        # Tab into the description area to fill some text
+        # Couldnt find a way to get into the box with selenium
+        driver.find_element_by_id("id_name").send_keys("\tThis is just a test. Be very afraid.")
+        # Select an option in the poroduct type
+        Select(driver.find_element_by_id("id_prod_type")).select_by_visible_text("Research and Development")
+        # "Click" the submit button to complete the transaction
+        driver.find_element_by_css_selector("input.btn.btn-primary").click()
+        # Query the site to determine if the product has been added
+        productTxt = driver.find_element_by_tag_name("BODY").text
+        # Assert ot the query to dtermine status of failure
+        # Also confirm success even if Product is returned as already exists for test sake
+        self.assertTrue(re.search(r'Product added successfully', productTxt) or
+            re.search(r'Product with this Name already exists.', productTxt))
+
+    def test_create_endpoint(self):
+        # Login to the site.
+        # Username and password will be gotten from environ
+        driver = self.login_page()
+        # Navigate to the Endpoint page
+        driver.get(self.base_url + "endpoint")
+        # "Click" the dropdown button to see options
+        driver.find_element_by_id("dropdownMenu1").click()
+        # "Click" the New Endpoint
+        driver.find_element_by_link_text("New Endpoint").click()
+        # Keep a good practice of clearing field before entering value
+        # Endpoints
+        driver.find_element_by_id("id_endpoint").clear()
+        driver.find_element_by_id("id_endpoint").send_keys("123.22.43.101\nmoving.com.rnd\nhttps://gorand.in/publish")
```

Comment:
Reduce this to one finding so that there is nothing left over after all tests have been run. This applies to the initially created product as well.

Reply:
I made sure product is created only once. By checking sure if i get assert error of already exist. then that is a success indicating that the test can move on. there is already a product to work to.

Reply:
endpoint finding reduced to one.